core/src/net/network_thread.rs

```diff
@@ -769,14 +771,46 @@ impl NetworkThread {
         }
         let rc = Rc::new(RefCell::new(channel));
         self.address_map.insert(address, rc.clone());
-        self.token_map.insert(self.token, rc);
+        self.token_map.put(self.token, rc);
+        self.check_connection_limit();
         self.next_token();
     }
 
-    /// Initiates a graceful closing sequence
+    /// Checks the current channel-count and initiates a graceful shutdown of the channel.
+    fn check_connection_limit(&mut self) -> () {
+        let channel_count = self.token_map.len() as i32;
+        let mut exceeded = channel_count - self.network_config.get_connection_limit() as i32;
```

Comment:
Whoa, integer overflow when `channel_count = 0` and `connection_limit > i32::MAX` (for example)!